container-runtime/src/Adapters/Systems/SystemForEmbassy/index.ts

```diff
@@ -199,11 +199,14 @@ export class SystemForEmbassy implements System {
   async execute(
     effects: HostSystemStartOs,
     options: {
+      id: string
       procedure: JsonPath
       input: unknown
       timeout?: number | undefined
     },
   ): Promise<RpcResult> {
+    effects = Object.create(effects)
+    effects.procedureId = options.id
```

Comment:
yeah unfortunately this doesn't work because you lose all the class methods if you do this, since they aren't actual fields on the object, they're part of the protoype